banyand/liaison/grpc/stream.go

```diff
@@ -76,18 +78,29 @@ func (s *streamService) Write(stream streamv1.StreamService_WriteServer) error {
 		}
 		if err != nil {
 			s.sampled.Error().Stringer("written", writeEntity).Err(err).Msg("failed to receive message")
-			reply(stream, s.sampled)
+			reply(nil, modelv1.Status_STATUS_RECEIVE_ERROR, 0, stream, s.sampled)
```

Comment:
Here is a bug introduced by me. When a non-EOF error comes, we should "return" instead of "continue".  The underlying issue causing the error is not addressed, and this could lead to further issues down the line.

Once you fixed this bug, "STATUS_RECEIVE_ERROR" is redundant as well.